### ISSUE TITLE
PoC support for Resource Owner Password Credential Flow

### ISSUE THREAD

**Issue by jsedanoj** [authts/oidc-client-ts#746]
Hello,

I need support for Resource Owner Password Credential Flow (https://www.rfc-editor.org/rfc/rfc6749#section-4.3). `oidc-client-ts` does not support it... but the changes needed to support it are very small.

I have made it in my OIDC playground, but I have needed to duplicate some classes in my project because they are @internal and cannot be used/extended in client projects. It would be much easier if this support was directly provided by oidc-client-ts. Watch my PoC at https://gitlab.com/javier-sedano/oidc-vue/-/tree/master/src/oidc-client-ts and deployed to "production" at https://jsedano.duckdns.org/oidc-vue/ .

My questions:

- Are you guys open to receiving such a contribution?
- How? Fork + PR?

BTW, I know that Resource Owner Password Credential Flow is only secure if the Client Application is fully trusted not to keep the credentials... but in scenarios where the IdP and the Client Application are owned by the same entity (which is my case) it is acceptable (for example, read https://www.rfc-editor.org/rfc/rfc6749#section-1.3.3 or  https://auth0.com/docs/get-started/authentication-and-authorization-flow/resource-owner-password-flow). So I would add the appropriate warning to the readme.

**Comment by pamapa** [comment]
Yes we are open for such an contribution. Would be nice if you could create a merge request within this project. I have looked at your code, the changes are mostly in the addition of a new manager function, i guess...

**Comment by jsedanoj** [comment]
Yes, one function and reusing a lot of already existing logic; mostly the only relevant change is a new POST to the IdP in TokenClient.

I have the change working and tested in my laptop, but cannot git push to the repo because I do not have permissions. Am I supposed to ask for contributor permissions? I understood "within this project" as a branch in this very same repo...

**Comment by pamapa** [comment]
> I have the change working and tested in my laptop, but cannot git push to the repo because I do not have permissions.
That is not how github works :-)

1. fork this project (via Web-UI), such that you have a copy for your own (with full permission)
2. create a new feature branch in your forked project (via git command or git-UI)
3. add a merge request in this project (via Web-UI) using the feature branch of your forked project

I quickly search for a tutorial maybe this helps you: https://www.youtube.com/watch?v=GZILYABgAoo

**Comment by jsedanoj** [comment]
Oh, no problem, I know how to do it, I just missunderstood your request.

Done, the PR is here: https://github.com/authts/oidc-client-ts/pull/747

**Comment by jsedano-emobg** [comment]
Any plan on when will we have a release including this feature?

The documentation in the README does not match the real functionality if you do `npm install oidc-client-ts`... because, well, it is not published.

**Comment by pamapa** [comment]
will do a release this week
